src/helpers/credential.rs

```diff
@@ -269,6 +279,138 @@ impl<'a> Credential<'a> {
+                id: "did:evan:zkp:placeholder_status#0".to_string(),
+                r#type: "RevocationList2020Status".to_string(),
+                revocation_list_index: "0".to_string(),
+                revocation_list_credential: "did:evan:zkp:placeholder_status".to_string(),
+            },
+        };
+        let unsigned_credential_str = serde_json::to_string(&unsigned_credential)?;
+        let nquads = convert_to_nquads(&unsigned_credential_str).await?;
+
+        let payload = OfferCredentialPayload {
+            issuer: subject_did_str_copy.to_string(),
+            subject: Option::from(subject_did_copy),
+            nquad_count: nquads.len(),
+        };
+        let offer_str = self
+            .vade_evan
+            .vc_zkp_create_credential_offer(
+                EVAN_METHOD,
+                TYPE_OPTIONS,
+                &serde_json::to_string(&payload)?,
+            )
+            .await
+            .map_err(|err| CredentialError::VadeEvanError(err.to_string()))?;
+
+        // Create credential request
+        let credential_values_str = serde_json::to_string(&credential_subject.data)?;
+        let request_str = self
+            .create_credential_request(
+                serde_json::to_string(&(issuer_public_key.as_str()))?.as_str(),
+                serde_json::to_string(bbs_secret)?.as_str(),
+                credential_values_str.as_str(),
+                offer_str.as_str(),
+                schema_did,
+            ).await?;
+        let (request, _key) : (BbsCredentialRequest, String) = serde_json::from_str(request_str.as_str())?;
+        let request_copy = request.clone();
+        let indices = vec![0];
+
+        // Issue credentials
+        let offer: BbsCredentialOffer = serde_json::from_str(offer_str.as_str())?;
+        let pload = IssueCredentialPayload {
+            unsigned_vc: unsigned_credential,
+            nquads,
+            issuer_public_key_id: r##"#bbs-key-1"##.to_string(),
+            issuer_public_key,
+            issuer_secret_key: bbs_secret.to_string(),
+            credential_request: request_copy,
+            credential_offer: offer,
+            required_indices: indices,
+        };
+        let pload_str = serde_json::to_string(&pload)?;
+        let credential_str = self
+            .vade_evan
+            .vc_zkp_issue_credential(EVAN_METHOD, TYPE_OPTIONS, &pload_str.as_str())
+            .await
+            .map_err(|err| CredentialError::VadeEvanError(err.to_string()))?;
+
+        // Finish credentials
+        let mut parsed_credential: Map<String, Value> = serde_json::from_str(credential_str.as_str())?;
+        parsed_credential.remove("proof");
+        let _credential_without_proof = serde_json::to_string(&parsed_credential)?;
+        let did_doc_nquads = convert_to_nquads(&_credential_without_proof).await?;
+        let credential: UnfinishedBbsCredential = serde_json::from_str(credential_str.as_str())?;
+        let credential_copy = credential.clone();
```

Comment:
Please use `.copy` where needed.

```suggestion
        let credential: UnfinishedBbsCredential = serde_json::from_str(credential_str.as_str())?;
```

Reply:
Done, thanks